### PR TITLE
Adding capability to configure GKE maintenance window

### PR DESCRIPTION
**Pull request by mishah334**
- Adding capability to configure GKE maintenance window
- Adding `deletion_policy` for `google_sql_user` to avoid failure during destroy.

Issue: https://github.com/astronomer/issues/issues/2906

### CODE REVIEW

File: db.tf [hunk 1]
```diff
@@ -63,8 +63,9 @@ resource "random_string" "postgres_airflow_password" {
...
 resource "google_sql_user" "airflow" {
-  count    = var.deploy_db ? 1 : 0
-  name     = "airflow"
-  instance = google_sql_database_instance.instance[0].name
-  password = local.postgres_airflow_password
+  count           = var.deploy_db ? 1 : 0
+  name            = "airflow"
+  deletion_policy = "ABANDON"
+  instance        = google_sql_database_instance.instance[0].name
+  password        = local.postgres_airflow_password
```
Comment (danielhoherd) [inline db.tf:70]: This change seems unrelated to the maintenance stuff. Is this something we should do in a different PR to make it easier to see and track? Seems like it is something that could be risky since it's managing the DB.

Comment (danielhoherd) [reply db.tf:70]: I see, makes sense. Thanks for clarifying that!

**Review by danielhoherd — APPROVED** [review]

**Review by shmanu017 — APPROVED** [review]
